### PR TITLE
CORE PROFILE: Removing the line width completely from gpu::lib

### PR DESCRIPTION
**Pull request by samcake**
Because it s not supporting wide lines correctly and universally anymore on core profile and we would rather have a specialized pipeline for this

So the idea is too remove the lineWidth from the Batch interface

Also turn the state "antialisedLineEnable" true by default

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: d07ab353fc69f815011814d6c5a463af2feaeb17 available at: https://gist.github.com/b8d13b04a3f82021f170<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5496.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5496.dmg

File: libraries/gpu/src/gpu/State.h [hunk 1]
```diff
@@ -260,7 +260,7 @@ class State {
             depthClampEnable(false),
             scissorEnable(false),
             multisampleEnable(false),
-            antialisedLineEnable(false),
+            antialisedLineEnable(true),
```
Comment (jherico) [inline libraries/gpu/src/gpu/State.h]: don't we need a glEnable(GL_LINE_SMOOTH) in the backendstate setup?

File: libraries/gpu/src/gpu/GLBackend.h [hunk 2]
```diff
@@ -445,7 +445,6 @@ class GLBackend : public Backend {
     void do_glUniformMatrix4fv(Batch& batch, uint32 paramOffset);
 
     void do_glColor4f(Batch& batch, uint32 paramOffset);
```
Comment (jherico) [inline libraries/gpu/src/gpu/GLBackend.h]: why is this still here?

**Comment by jherico** [comment]
for places where we've removed the line width of something other than 1.0 we should add a FIXME to restore the current rendering width.

File: libraries/gpu/src/gpu/State.h (hunk 1 above)
Comment (samcake) [inline libraries/gpu/src/gpu/State.h]: THis is done as part of the regular state cached values line 587 of GLBackendState.cpp
void GLBackend::do_setStateAntialiasedLineEnable(bool enable) {

**Comment by hifi-gustavo** [comment]
Build console output for commit: 34183b9ddf54541e84eea4fbad6b2cf902a00dc0 available at: https://gist.github.com/b8d13b04a3f82021f170<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5496.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5496.dmg

**Pull request merged**